modules/wallet/broadcast.go

```diff
@@ -0,0 +1,74 @@
+package wallet
+
+import (
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+// broadcastedTSet is a helper struct to keep track of transaction sets and to
+// help rebroadcast them.
+type broadcastedTSet struct {
+	firstTry     types.BlockHeight            // first time the tSet was broadcasted
+	lastTry      types.BlockHeight            // last time the tSet was broadcasted
+	confirmedTxn map[types.TransactionID]bool // tracks confirmed txns of set
+	transactions []types.Transaction          // the tSet
+	id           modules.TransactionSetID     // the tSet's ID
+	w            *Wallet
+}
+
+// persistBTS is the on-disk version of the broadcastedTSets structure. This is
+// necessary since we can't marshal a map directly. Instead we make sure that
+// confirmedTxn[i] corresponds to the confirmation state of transactions[i].
+type persistBTS struct {
+	FirstTry     types.BlockHeight   // first time the tSet was broadcasted
+	LastTry      types.BlockHeight   // last time the tSet was broadcasted
+	ConfirmedTxn []bool              // tracks confirmed txns of set
+	Transactions []types.Transaction // the tSet
+}
+
+// confirmed is a helper function that sets a certain transactions to confirmed
+// or unconfirmed. It also updates the state on disk.
+func (bts *broadcastedTSet) confirmed(txid types.TransactionID, confirmed bool) error {
```

Comment:
this function name suggests a query, as though you are asking for the confirmed status of a transaction. Should change the name to 'markConfirmation' or something similar.